extensions/s3/src/main/java/io/deephaven/extensions/s3/S3ChannelContext.java

```diff
@@ -474,12 +474,12 @@ private ByteBuffer getFullFragment() throws ExecutionException, InterruptedExcep
             // apiCallTimeout.
             final long readNanos = instructions.readTimeout().plusMillis(100).toNanos();
             final Boolean isComplete = consumerFuture.get(readNanos, TimeUnit.NANOSECONDS);
-            if (!isComplete) {
+            if (Boolean.FALSE.equals(isComplete)) {
                 throw new UncheckedDeephavenException(String.format("Failed to complete request %s", requestStr()));
             }
             final ByteBuffer result = get();
             if (result == null) {
-                throw new IllegalStateException(
+                throw new UncheckedDeephavenException(
```

Comment:
I think in general we want to `localProducer.complete(false);` when we have no buffer in the `Sub`.
`IllegalStateException` is reasonable if we try to `fill` from a `Request` that is provably not acquired (since its buffer is `null`).

Reply:
```
            final Boolean isComplete = consumerFuture.get(readNanos, TimeUnit.NANOSECONDS);
            if (!Boolean.TRUE.equals(isComplete)) {
                throw new IllegalStateException(String.format("Failed to complete request %s", requestStr()));
            }
            final ByteBuffer result = get();
            if (result == null) {
                throw new IllegalStateException(
                        String.format("Tried to read from no-longer-acquired Request, %s", requestStr()));
            }
```